source/guide_notify_push.rst

```diff
@@ -0,0 +1,181 @@
+.. highlight:: console
+.. author:: Marius Bertram <marius@brtrm.de>
+
+
+.. tag:: lang-rust
+.. tag:: web
+.. tag:: file-storage
+.. tag:: sync
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/nextcloud.svg
+      :align: center
+
+###########
+notify_push
+###########
+
+.. tag_list::
+
+notify_push is a backend service to inform Nextcloud clients on file changes.
+The default behavior of the client is to periodically, at short intervals, request changes from the server.
+This results in a large proportion of the server load.
+With notify_push the requests can be greatly reduced.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :lab:`Nextcloud <guide_nextcloud>`
+  * :lab:`redis <guide_redis>`
+  * :manual:`domains <web-domains>`
+  * :manual:`web backends <web-backends>`
+  * :manual:`supervisord <daemons-supervisord>`
+
+Prerequisites
+=============
+
+We need a running Nextcloud instance with  version >= 21.
+With configured redis.
+
+Installation
+============
+
+Install Client Push App in Nextcloud
+------------------------------------
+We need to install the client_push_app from the Nextcloud AppStore.
+
+`Client Push`_
+
+Download Binary
+---------------
+Download the latest release and make it executable:
+
+.. code-block:: console
+
+    [isabell@stardust ~]$ curl -L https://github.com/nextcloud/notify_push/releases/latest/download/notify_push-x86_64-unknown-linux-musl -o ${HOME}/bin/notify_push
+        % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                    Dload  Upload   Total   Spent    Left  Speed
+        100   169  100   169    0     0    401      0 --:--:-- --:--:-- --:--:--   402
+    [isabell@stardust ~]$ chmod u+x ${HOME}/bin/notify_push
+    [isabell@stardust ~]$
```

Comment:
No the app is just a connector to the binary backend. Indeed we can just install the connector app via `occ`.

Reply:
In my opinion it would be more consistent to download the binary yourself.

Reply:
> In my opinion it would be more consistent to download the binary yourself.

Why?

If you do the update by just downloading the binary file the compatibility of the `latest` version to the different Nextcloud release branches may not be guaranteed in future releases. The Nextcloud company usually delays the updates to the latest server releases for most users so a manual download to a latest notify app version may not work. The `occ app:update --all` command ensures that the apps only update to compatible versions (as they are tagged). 

On uberspace the last lines of the [`nextcloud-update`](https://lab.uberspace.de/guide_nextcloud.html#updates)  script would look like this:
```
[...]
php ~/html/occ app:update --all
php ~/html/occ maintenance:mode --off
/usr/sbin/restorecon -R ~/html
supervisorctl restart notify_push
```